projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -19,98 +17,78 @@ import {
 import { DateFormControlService } from './providers/date-form-control.service';
 import { DateIOService } from './providers/date-io.service';
 import { DateNavigationService } from './providers/date-navigation.service';
 import { DatepickerEnabledService } from './providers/datepicker-enabled.service';
 import { DatepickerFocusService } from './providers/datepicker-focus.service';
-import { datesAreEqual } from './utils/date-utils';
 
 // There are four ways the datepicker value is set
 // 1. Value set by user typing into text input as a string ex: '01/28/2015'
 // 2. Value set explicitly by Angular Forms APIs as a string ex: '01/28/2015'
 // 3. Value set by user via datepicker UI as a Date Object
 // 4. Value set via `clrDate` input as a Date Object
 
-@Directive({
-  selector: '[clrDate]',
-  host: {
-    '[class.clr-input]': 'true',
-  },
-  providers: [DatepickerFocusService],
-})
-export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implements OnInit, AfterViewInit, OnDestroy {
+@Directive()
+export class ClrDateInputBase extends WrappedFormControl<ClrDateContainer> implements OnInit, OnDestroy {
   static ngAcceptInputType_date: Date | null | string;
 
   @Input() placeholder: string;
-  @Output('clrDateChange') dateChange = new EventEmitter<Date>(false);
 
   protected override index = 1;
 
-  private initialClrDateInputValue: Date;
-  private previousDateChange: Date;
-
   constructor(
     viewContainerRef: ViewContainerRef,
     injector: Injector,
     protected override el: ElementRef<HTMLInputElement>,
     protected override renderer: Renderer2,
+    @Inject(DOCUMENT) protected document: any,
     @Self()
     @Optional()
     protected control: NgControl,
     @Optional() private container: ClrDateContainer,
-    @Optional() private dateIOService: DateIOService,
-    @Optional() private dateNavigationService: DateNavigationService,
+    @Optional() protected dateIOService: DateIOService,
+    @Optional() protected dateNavigationService: DateNavigationService,
     @Optional() private datepickerEnabledService: DatepickerEnabledService,
     @Optional() private dateFormControlService: DateFormControlService,
     @Inject(PLATFORM_ID) private platformId: any,
     @Optional() private focusService: FocusService,
-    private datepickerFocusService: DatepickerFocusService
+    protected datepickerFocusService: DatepickerFocusService
   ) {
     super(viewContainerRef, ClrDateContainer, injector, control, renderer, el);
   }
 
-  @Input('clrDate')
-  set date(date: Date | string) {
-    if (this.previousDateChange !== date) {
-      this.updateDate(this.getValidDateValueFromDate(date as Date));
-    }
-
-    if (!this.initialClrDateInputValue) {
-      this.initialClrDateInputValue = date as Date;
-    }
-  }
-
   @Input()
   set min(dateString: string) {
-    this.dateIOService.setMinDate(dateString);
-    this.triggerControlValidation();
+    if (!this.dateNavigationService.isRangePicker) {
```

Comment:
Why isn't this in the `ClrDateInput` directive. It only applies to that directive.